### PR TITLE
Remove context from distance formatter options

### PR DESCRIPTION
**Pull request by Lebedinsky**
### Description
This PR is about removing `context` from `DistanceFormatterOptions` #3939 .

### Changelog
```
<changelog>Removed context from DistanceFormatterOptions</changelog>
```

### CODE REVIEW

**Comment by LukasPaczos** [comment]
Could you update the PR to make the tests pass?

**Comment by truburt** [comment]
Fixes #3939

File: libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/model/DistanceRemainingFormatter.kt [hunk 1]
```diff
@@ -8,13 +9,15 @@ import com.mapbox.navigation.ui.base.formatter.ValueFormatter
 /**
  * Formats trip related data for displaying in the UI
  *
+ * @param applicationContext from which to get localized strings from
  * @param distanceFormatterOptions to build the [DistanceRemainingFormatter]
  */
 class DistanceRemainingFormatter(
+    private val applicationContext: Context,
```
Comment (kmadsen) [inline libnavui-tripprogress/src/main/java/com/mapbox/navigation/ui/tripprogress/model/DistanceRemainingFormatter.kt:16]: this defeats the purpose of removing it from `DistanceFormatterOptions`

**Comment by kmadsen** [comment]
Solution needs broader discussion https://github.com/mapbox/mapbox-navigation-android/pull/4135#discussion_r616212535

It is better to leave the context inside the DistanceFormatterOptions, than to take it out of the options for without a tangible reason.

**Pull request closed without merging**